### PR TITLE
SPR-49 Frontend Main Page

### PR DESCRIPTION
**Pull request by dbarkowsky**
# Description

This PR includes the following proposed change(s):

- Basic frontend page started
- Includes a table that retrieves records from the API (no keycloak yet, so TESTING env must be true)
- Some gov't-styled components (header, buttons)

Only for local use so far. Will prepare for Keycloak and cloud deployment in future PR.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change
- [x] Dependencies added/removed

## Aspect(s) of Project Affected
- [x] Frontend
- [x] API
- [ ] Database
- [ ] Workflows
- [ ] Documentation

## Best Practices Checklist
- [x] I have performed a self-review of my own code
- [x] My code follows the Airbnb React Style Guidelines
- [x] Documentation has been updated to reflect my changes
- [x] New and existing tests pass locally with my changes

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Tested locally

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -11,8 +11,14 @@
     "cypress:e2e": "cypress run --e2e"
   },
   "dependencies": {
+    "@bcgov/bc-sans": "^1.0.1",
```
Comment (akroon3r) [inline app/package.json:14]: apparently this is being updated to v2 this week? watch out for it. @Bottle7 @BradyMitch

File: app/src/components/bcgov/ButtonStyles.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    borderRadius: '4px',
+    padding: '12px 24px',
+    textAlign: 'center',
+    textDecoration: 'none',
+    display: 'block',
+    fontSize: '14px',
+    fontWeight: '700',
+    letterSpacing: '1px',
+    cursor: 'pointer',
+  '&:hover': {
```
Comment (akroon3r) [inline app/src/components/bcgov/ButtonStyles.ts:15]: this seems like you're missing an indent here? looks weird to me.

Comment (dbarkowsky) [reply app/src/components/bcgov/ButtonStyles.ts:15]: There's an extra indent for the non-hover styling. Was trying to keep it visually identifiable, but can align it all if you think that's better.

File: app/package.json (hunk 1 above)
Comment (dbarkowsky) [reply app/package.json:14]: Any way to find out if it will be breaking? I don't see any update info on their repo.

Comment (akroon3r) [reply app/package.json:14]: doesn't appear to be breaking changes from the inside information that I'm reading, but I'm not sure if the communication is finalized or official yet. just keep an eye out.

File: app/src/components/bcgov/ButtonStyles.ts (hunk 2 above)
Comment (Bottle7) [reply app/src/components/bcgov/ButtonStyles.ts:15]: I typically use an extra line break. Indentation is finicky and most of the time our prettier config will fix indentations like this.

Comment (dbarkowsky) [reply app/src/components/bcgov/ButtonStyles.ts:16]: Updated to use consistent indentation with line break instead.

**Review by akroon3r — APPROVED** [review]
I think you're good based on the update to feedback you just pushed.

**Comment by BradyMitch** [comment]
Since you're using npm i in the Dockerfile, theres no need to add package-lock.json to the repo. It gets generated based on package.json so I would remove it and gitignore it

**Pull request merged**